lib/_http_server.js

```diff
@@ -189,7 +189,7 @@ ServerResponse.prototype.writeHead = function(statusCode, reason, obj) {
 
   statusCode |= 0;
   if (statusCode < 100 || statusCode > 999)
-    throw new RangeError(`Invalid status code: ${statusCode}`);
+    throw new RangeError(`Invalid status code: ${this.statusCode}`);
```

Comment:
Speaking from experience, that's confusing to an end user, because it ends up as 0 for most invalid input. (In my case, someone had been setting error.code to a custom status code, but then an ENOENT error caused "RangeError: Invalid status code: 0", and I spent a good part of a day tracking it down because I was looking for the wrong thing.)

However, you do have a valid point, so what about a couple of alternatives - either include both the coerced value and the original value in the message, or else have a type check earlier that throws on non-Number input with the original value & type?